### DISCUSSION TITLE
boundary condition and output questions

### DISCUSSION THREAD

**Discussion by dshen1993** [CliMA/Oceananigans.jl#3613] [Computational science!]
I want to set the gradient boundary condition to east, west, south, north directions in u and v variables. However, for u variable, cannot specify east boundary condition using GradientBoundaryCondition at Face(). So, how should i solve this problem. 

Then, i have another question. i want to output wind stress (top flux boundary condition) to jld2 or netcdf. But i cannot save or find wind stress (top flux boundary condition) from model. So, which field or variable can i set to save wind stress. Is there a simple example?

Thanks.

**Comment by glwagner** [comment]
You need to use `Open` to set boundary-normal velocities. However, what you're trying to do may not be supported. @jagoosw may know more. Can you please provide more detail about the setup you are trying to run?

> Then, i have another question. i want to output wind stress (top flux boundary condition) to jld2 or netcdf. But i cannot save or find wind stress (top flux boundary condition) from model. So, which field or variable can i set to save wind stress. Is there a simple example?

This depends on how you are setting the wind stress --- are you using a function or array? If you provide more details we can figure it out.

I converted this to a discussion because we are not trying to change the source code (as far as I can tell).

**Reply by dshen1993** [reply]
Thanks for your response. I want to make a simulation under hurricane. I solve the wind stress problem using kernelfunctionoperation. But I still have some questions about the horizontal boundary conditions. We can set periodic boundary conditions in x and y directions. But i want to set all horizontal bounadries (south, north, east, west) to radiation boundary condition, so is it possible to using open boundary condition with a function to present the radiation boundary condition?

**Reply by glwagner** [reply]
Not without implementing it yourself. @jagoosw is working on this in #3482 .

In order to help further you need to explicitly give us the code you are trying to use.
